WcaOnRails/app/views/competitions/_competition_form.html.erb

```diff
@@ -250,21 +248,19 @@
 
         <script>
           function UpdateDues(){
-
-            var obj = $("#competition-form").serializeJSON();
-            // Coerce compLimitEnabled to a bool, as it is a string obj['competition[competitor_limit_enabled]'] returns a string
-            var compLimitEnabled = ("true" === obj['competition[competitor_limit_enabled]'])
-            var compLimit = compLimitEnabled ? obj['competition[competitor_limit]'] : null
-            var currencyCode = obj['competition[currency_code]']
-
+            var compLimitEnabled = $("#competition_competitor_limit_enabled").val();
```

Comment:
You are missing the previous comparison to `"true"` to make sure that it's coerced into a boolean on the JS side. If you just use `.val()` there is a risk that this `var compLimitEnabled` contains a **string** that reads `"true"` or `"false"` which is dangerous because `if ("false") console.log("foo") else console.log("bar")` logs `foo`. Non-empty strings are truthy in JS, even if the string reads "false".

For reference, the previous code was
```js
// Coerce compLimitEnabled to a bool, as it is a string obj['competition[competitor_limit_enabled]'] returns a string
var compLimitEnabled = ("true" === obj['competition[competitor_limit_enabled]'])
```

Reply:
Done, thanks for pointing it out, @gregorbg.